snowpack/src/types/snowpack.ts

```diff
@@ -11,15 +11,20 @@ export type DeepPartial<T> = {
 
 export type EnvVarReplacements = Record<string, string | number | true>;
 
-export type SnowpackBuiltFile = {code: string | Buffer; map?: string};
+export type FileContents = string | Buffer
+
+export type SnowpackBuiltFile = {
+  contents: FileContents;
+  map?: string
+};
 export type SnowpackBuildMap = Record<string, SnowpackBuiltFile>;
 
 /** Standard file interface */
 export interface SnowpackSourceFile {
   /** base extension (e.g. `.js`) */
   baseExt: string;
   /** file contents */
-  contents: string;
+  contents: FileContents;
```

Comment:
Nice catch! This was causing issues.

Reply:
WDYT about this change?

```diff
- export interface SnowpackSourceFile {
+export interface SnowpackSourceFile<Type = FileContents> {

-   contents: FileContents;
+  contents: Type;
```

That way `SnowpackSourceFile` can be used as before with a `string | Buffer` type, or it can be used like `SnowpackSourceFile<string>` in cases where we’re confident about the encoding type? It could be the best of both—allows for polymorphic types and ensure we have to check scenarios, but also lets us prevent noise in other places.